02-intermediate-ruby/references-and-values.md

```diff
@@ -0,0 +1,267 @@
+# References and Values
+
+## Learning Goals
+At the end of this lesson, students should be able to...
+
+- Discuss how objects are stored in Ruby
+- Differentiate between _references_ and _values_
+- Compare _modifying an object_ with _reassigning a variable_
+
+## Motivation
+
+We're going to start today with some Ruby code that does something a little unexpected. It's a method that takes an array of strings as an argument, and truncates (chops off the end of) all the strings with more than three characters. Or at least that's the idea.
+
+```ruby
+def short_strings_1(input)
+  result = []
+  input.each do |word|
+    # Slice characters 0 to 2
+    result << word[0..2]
+  end
+  input = result
+end
+
+pets = ['dog', 'parrot', 'cat', 'llama']
+short_strings_1(pets)
+puts "#{pets}"
+
+```
+
+Running this code results in `["dog", "parrot", "cat", "llama"]`. The array in unchanged! Let's do some debugging:
+
+```ruby
+def short_strings_1(input)
+  result = []
+  input.each do |word|
+    # Slice characters 0 to 2
+    result << word[0..2]
+  end
+  input = result
+  puts "Inside short_strings_1, input is"
+  puts "#{input}"
+end
+
+pets = ['dog', 'parrot', 'cat', 'llama']
+short_strings_1(pets)
+puts "After calling short_strings_1"
+puts "#{pets}"
+```
+
+The results:
+
+```
+Inside short_strings_1, input is
+["dog", "par", "cat", "lla"]
+After calling short_strings_1
+["dog", "parrot", "cat", "llama"]
+```
+
+Seems like our method is indeed creating a list of shortened words, but our outer variable isn't being updated. The reason why has to do with references and values, and how data is stored in a computer.
```

Comment:
Can you update this sentence to reference the specific variables used in the method and outside the method